src/serializers/addresses-serializer.js

```diff
@@ -0,0 +1,75 @@
+import { Serializer as JsonApiSerializer } from 'jsonapi-serializer';
+import _ from 'lodash';
+
+import { serializerOptions } from 'utils/jsonapi';
+import { openapi } from 'utils/load-openapi';
+import { apiBaseUrl, resourcePathLink, paramsLink } from 'utils/uri-builder';
+
+const addressResourceProp = openapi.components.schemas.AddressResult.properties.data.properties;
+const addressResourceType = addressResourceProp.type.enum[0];
+const addressResourceAttributes = addressResourceProp.attributes.allOf;
+const addressCombinedAttributes = _.merge(
+  addressResourceAttributes[0], addressResourceAttributes[1],
+);
+const addressResourceKeys = _.keys(addressCombinedAttributes.properties);
+
+/**
+ * Some fields need to be massaged before the can be passed to the serializer
+ *
+ * @param {object} rawAddresses raw address data from data source
+ */
+const prepareRawData = (rawAddresses) => {
+  // handle sub objects programmatically
+  _.forEach(rawAddresses, (address) => {
+    _.forEach(address, (value, key) => {
+      const splitKey = key.split('.');
+      if (splitKey.length > 1) {
+        if (!address[splitKey[0]]) {
+          address[splitKey[0]] = {};
+        }
+        address[splitKey[0]][splitKey[1]] = value;
+      }
+    });
+  });
+};
```

Comment:
Handling sub-objects in the jobs endpoint took a while. This is one way it can be done programmatically if you name the fields in the contrib correctly. What are your thoughts? If you like it we can improve on it and add it to the skeleton.

Reply:
Agreed, I think that format makes the most sense.